packages/redux-dynostore-react-redux/README.md

```diff
@@ -44,6 +44,30 @@ export default dynamic('identifier', somethingDynamic(), { context: CustomReactR
 
 This option overrides the `context` used for accessing the store in the React context.  If you are overriding this value, you should also be overriding the context passed to any `Provider` and/or `connect` components as well.  Please refer to the [Redux documentation](https://react-redux.js.org/using-react-redux/accessing-store#providing-custom-context) for more details on this use case.
 
+### `DynamicProvider`
+
+`dynamic` components will not render when server-side rendering (SSR) your app unless you add a `DynamicProvider` somewhere near the root of of your component tree (anywhere above the the first `dynamic` component).
+This will allow the component to call their enhancers synchronously in the first render pass.  You must also ensure that the `DynamicProvider` is also rendered when hydrating that app on the client.
+
+```js
+import ReactDom from 'react'
+import { Provider } from 'react-redux'
+import dynamic, { DynamicProvider } from '@redux-dynostore/react-redux'
+
+import store from './store'
+import MyDynamicComponent from './MyDynamicComponent
+
+ReactDom.render((
+  <Provider store={store}>
+    <DynamicProvider>
+      <MyDynamicComponent />
+    </DynamicProvider>
+  </Provider>
+), document.getElementById('root'))
+```
+
+While not required for client-side rendering (CSR), `DynamicProvider`, it can also slightly improve performance of the first render pass.  It's use in CSR is generally not advised and not a supported use case, so use it at your own risk.
```

Comment:
Do you think this wording might be confusing? It has better performance but its use is not advised?

Reply:
Yeah, it felt weird writing it.  I wanted to convey that it might have some unintended side-effects in CSR (might also in SSR, but 🤷‍♂️) but there is still a performance benefit if you don't get any bugs from using it (I think there would only be any in a concurrent rendering world, but I just don't know).

What do you think? Leave mentions of CSR it out all together or drop the warning and assume all will be fine (YOLO)?

Reply:
Maybe word it in this kind of way?

> that being said, due to limited testing, there might be unforeseen impacts of using this provider, so use it at your own risk.

Reply:
Would you say that's true for SSR as well, or only CSR?

Reply:
Well, if you're using it for SSR, you're also using it for hydration, so I don't see them as being distinct. 

Honestly I doubt there will be issues with the implementation, but it's still probably wise to communicate on the limited testing that we did.

Reply:
My only concern is that the synchronous update causes a warning normally and we're only comfortable with it in the first render because `react` _doesn't_ warn about it.  It is possible that in a future concurrent rendering world, the first pass might not be safe either and we will have to revisit it.